src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,347 @@
+    void HandlePauseAction(HandlerContext & ctx, const Commands::PauseAction::DecodableType & commandData);
+    void HandlePauseActionWithDuration(HandlerContext & ctx, const Commands::PauseActionWithDuration::DecodableType & commandData);
+    void HandleResumeAction(HandlerContext & ctx, const Commands::ResumeAction::DecodableType & commandData);
+    void HandleEnableAction(HandlerContext & ctx, const Commands::EnableAction::DecodableType & commandData);
+    void HandleEnableActionWithDuration(HandlerContext & ctx,
+                                        const Commands::EnableActionWithDuration::DecodableType & commandData);
+    void HandleDisableAction(HandlerContext & ctx, const Commands::DisableAction::DecodableType & commandData);
+    void HandleDisableActionWithDuration(HandlerContext & ctx,
+                                         const Commands::DisableActionWithDuration::DecodableType & commandData);
+};
+
+class Delegate
+{
+public:
+    virtual ~Delegate() = default;
+
+    /**
+     * Get the action at the Nth index from list of actions.
+     * @param index The index of the action to be returned. It is assumed that actions are indexable from 0 and with no gaps.
+     * @param action A reference to the GenericActionStruct which copies the buffer into it's own memory.
+     *               Name field (CharSpan) of ActionStruct contails the name of an action.
+     *               The API makes a copy of name.
+     * @return Returns a CHIP_NO_ERROR if there was no error and the action was returned successfully.
+     * CHIP_ERROR_PROVIDER_LIST_EXHAUSTED if the index in beyond the list of available actions.
+     */
+    virtual CHIP_ERROR ReadActionAtIndex(uint16_t index, GenericActionStruct & action);
+
+    /**
+     * Get the EndpointList at the Nth index from list of endpointList.
+     * @param index The index of the endpointList to be returned. It is assumed that actions are indexable from 0 and with no gaps.
+     * @param epList A reference to the GenericEndpointList which copies the buffer into it's own memory.
+     *               Name field (CharSpan) of EndpointList contails the name of an endpoint list.
+     *               The API makes a copy of name.
+     * @return Returns a CHIP_NO_ERROR if there was no error and the epList was returned successfully.
+     * CHIP_ERROR_PROVIDER_LIST_EXHAUSTED if the index in beyond the list of available endpointList.
+     */
+    virtual CHIP_ERROR ReadEndpointListAtIndex(uint16_t index, GenericEndpointList & epList);
+
+    /**
+     * Find the action with matching actionId in the list of action.
+     * @param actionId The action to be find in the list of action.
+     * @return Returns a CHIP_NO_ERROR if matching action is found.
+     * CHIP_ERROR_NOT_FOUND if the matching action does not found in the list of action.
+     */
+    virtual CHIP_ERROR FindActionIdInActionList(uint16_t actionId);
```

Comment:
This would be better as an actual boolean predicate, instead of using CHIP_ERROR to communicate a boolean, no?